src/dotnet-new3/New3CommandFactory.cs

```diff
@@ -27,12 +27,41 @@ internal static Command Create()
                 CommandName,
                 (ParseResult parseResult) => HostFactory.CreateHost(parseResult.GetValueForOption(_debugDisableBuiltInTemplatesOption)),
                 (ParseResult parseResult) => new TelemetryLogger(null, parseResult.GetValueForOption(_debugEmitTelemetryOption)),
-                new NewCommandCallbacks());
+                new NewCommandCallbacks()
+                {
+                    RestoreProject = (projectPath) => ExecuteDotnetCommand(Dotnet.Restore(projectPath)),
+                    AddProjectReference = (projectPath, references) => ExecuteDotnetCommand(Dotnet.AddProjectToProjectReference(projectPath, references.ToArray())),
+                    AddPackageReference = (projectPath, package, version) => ExecuteDotnetCommand(Dotnet.AddPackageReference(projectPath, package, version)),
+                    AddProjectsToSolution = (solutionPath, projects, solutionFolder) => ExecuteDotnetCommand(Dotnet.AddProjectsToSolution(solutionPath, projects, solutionFolder)),
+                });
 
             newCommand.AddGlobalOption(_debugEmitTelemetryOption);
             newCommand.AddGlobalOption(_debugDisableBuiltInTemplatesOption);
             newCommand.AddCommand(new CompleteCommand());
             return newCommand;
         }
+
+        private static bool ExecuteDotnetCommand(Dotnet command)
+        {
+            command.CaptureStdOut();
+            command.CaptureStdErr();
+            Console.WriteLine($"Running '{command.Command}':");
+            var result = command.Execute();
+            Console.WriteLine(result.ExitCode == 0 ? "Command succeeded." : "Command failed.");
+            WriteCommandOutput(result);
+            return result.ExitCode == 0;
+        }
+
+        /// <summary>
+        /// Writes formatted command output from <paramref name="process"/>.
+        /// </summary>
+        private static void WriteCommandOutput(Dotnet.Result process)
```

Comment:
Why do we want all this output in `Console`?

Reply:
`dotnet-new3` doesn't have access to `Reporter`.